rtengine/simpleprocess.cc

```diff
@@ -1369,41 +1369,75 @@ class ImageProcessor {
             params.crop.enabled = false;
         }
 
-        if (params.resize.enabled) {
-            // resize image
-            LabImage *resized = new LabImage(imw, imh);
-            ipf.Lanczos(tmplab, resized, tmpScale);
-            delete tmplab;
-            tmplab = resized;
-            params.resize.enabled = false;
+        assert(params.resize.enabled);
 
-            params.sharpening = params.prsharpening;
-            params.impulseDenoise.thresh = 
-                int(float(params.impulseDenoise.thresh) * tmpScale);
-            if (tmpScale < 0.5) {
-                params.impulseDenoise.enabled = false;
-            }
-            params.wavelet.strength =
-                int(float(params.wavelet.strength) * tmpScale);
-            params.dirpyrDenoise.luma *= tmpScale;
-            //params.dirpyrDenoise.smethod = "shal";
-
-            const double dirpyreq_scale = min(tmpScale * 1.5, 1.0);
-            for (int i = 0; i < 6; ++i) {
-                auto &n = params.dirpyrequalizer.mult[i];
-                n = 1.0 + (n - 1.0) * dirpyreq_scale;
-            }
+        // resize image
+        LabImage *resized = new LabImage(imw, imh);
+        ipf.Lanczos(tmplab, resized, tmpScale);
+        delete tmplab;
+        tmplab = resized;
+
+        adjust_procparams(tmpScale);
             
-            fw = imw;
-            fh = imh;
-        }
+        fw = imw;
+        fh = imh;
 
         delete baseImg;
         baseImg = new Imagefloat(fw, fh);
         ipf.lab2rgb(*tmplab, *baseImg, params.icm.working);
         delete tmplab;
     }
 
+    void adjust_procparams(double scale_factor)
+    {
+        procparams::ProcParams& params = job->pparams;
+        procparams::ProcParams defaultparams;
+
+#define ADJUST_RADIUS_(n, f)                                              \
+        do {                                                            \
+            auto delta = (params. n - defaultparams. n) * f;            \
+            params. n = defaultparams. n + delta;                       \
+        } while (false)
+        
+        params.resize.enabled = false;
+
+        if (params.prsharpening.enabled) {
+            params.sharpening = params.prsharpening;
+        } else {
+            ADJUST_RADIUS_(sharpening.radius, scale_factor);
+        }
+        params.impulseDenoise.thresh = 
+            int(float(params.impulseDenoise.thresh) * scale_factor);
+        if (scale_factor < 0.5) {
+            params.impulseDenoise.enabled = false;
+        }
+        params.wavelet.strength =
+            int(float(params.wavelet.strength) * scale_factor);
```

Comment:
> I tend to do explicit type conversions because sometimes I have to work with the Microsoft C++ compiler and I like to avoid this annoying warning.

Yep, that can be a PITA. But in this case it's even suboptimal, as you first explicitly convert to `float` which is then converted to a `double` implicitly.

> I suppose RT doesn't care about the MS compiler anyway...

Me, too. :smile: 

But we do care about Clang.